docs/architecture/engine.md

```diff
@@ -0,0 +1,270 @@
+# Storage Engine Overview
+
+M3DB is a time series Database that was primarily designed to be horizontally scalable and handle a large volume of monitoring time series data.
+
+## Caveats / Limitations
+
+M3DB currently does not support the following:
+
+1. Any form of indexing. While M3DB is useful as a horizontally scalable datastore that provides extremely high compression ratios for time series data, it does not currently perform any type of indexing. This feature is currently under development and future versions of M3DB will have support for a built-in index that can be used standalone for smaller M3DB clusters (up to 10 nodes).
+2. Updates / deletes. All data written to M3DB is immutable.
+3. Writing very far into the past and future. M3DB was originally designed for storing high volumes of monitoring data, and thus writing data at arbitrary timestamps in the past and future was never an intended design goal. That said, this feature is currently under development and future versions of M3DB will have support for this.
```

Comment:
Don't wanna use the word configurable because its confusing in the sense that most databases allow you to write any data any time. I'm gonna replace it with "arbitrarily" and then make it less defensive if that works